ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml

```diff
@@ -0,0 +1,630 @@
+
+        cd /tmp/tf
+        echo "Creating infra"
+        terraform init -input=false -no-color &
+        wait "$!"
+
+        terraform apply -auto-approve -no-color &
+        wait "$!"
+
+        echo "Waiting for bootstrap to complete"
+        openshift-install --dir=/tmp/artifacts/installer wait-for bootstrap-complete &
+        wait "$!"
+
+        echo "Bootstrap complete, destroying bootstrap VM"
+        terraform apply -auto-approve -var 'bootstrap_complete=true' -no-color &
+        wait "$!"
+
+        echo "Completing UPI setup"
+        openshift-install --dir=/tmp/artifacts/installer wait-for install-complete &
+        wait "$!"
+
+    # Performs cleanup of all created resources
+    - name: teardown
+      image: ${LOCAL_IMAGE_UPI_INSTALLER}
+      terminationMessagePolicy: FallbackToLogsOnError
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-profile
+        mountPath: /etc/openshift-installer
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      env:
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      - name: TYPE
+        value: ${CLUSTER_TYPE}
+      - name: KUBECONFIG
+        value: /tmp/artifacts/installer/auth/kubeconfig
+      - name: AWS_SHARED_CREDENTIALS_FILE
+        value: /etc/openshift-installer/.awscred
+      - name: TFVARS_PATH
+        value: /etc/openshift-installer/secret.auto.tfvars
+      - name: AWS_REGION
+        value: us-east-1
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        function queue() {
+          local TARGET="${1}"
+          shift
+          local LIVE="$(jobs | wc -l)"
+          while [[ "${LIVE}" -ge 45 ]]; do
+            sleep 1
+            LIVE="$(jobs | wc -l)"
+          done
+          echo "${@}"
+          if [[ -n "${FILTER}" ]]; then
+            "${@}" | "${FILTER}" >"${TARGET}" &
+          else
+            "${@}" >"${TARGET}" &
+          fi
+        }
+
+        function teardown() {
+          set +e
+          touch /tmp/shared/exit
+          export PATH=$PATH:/tmp/shared
+
+          echo "Gathering artifacts ..."
+          mkdir -p /tmp/artifacts/pods /tmp/artifacts/nodes /tmp/artifacts/metrics /tmp/artifacts/bootstrap /tmp/artifacts/network
+
+          if [ -f /tmp/artifacts/installer/.openshift_install_state.json ]
+          then
+              # Remove VSPHERE_USER and VSPHERE_PASSWORD from install state json
+              export VSPHERE_USER=$(grep -oP 'vsphere_user="\K[^"]+' ${TFVARS_PATH})
+              export VSPHERE_PASSWORD=$(grep -oP 'vsphere_password="\K[^"]+' ${TFVARS_PATH})
+              sed -i "s;${VSPHERE_USER};REDACTED;g" /tmp/artifacts/installer/.openshift_install_state.json
+              sed -i "s;${VSPHERE_PASSWORD};REDACTED;g" /tmp/artifacts/installer/.openshift_install_state.json
+
+              # bootstrap.ign also contains passwords
+              rm -rf /tmp/artifacts/installer/bootstrap.ign
```

Comment:
If we are going to remove the bootstrap.ign file, then we should also remove the `*bootstrap.Bootstrap` asset from `.openshift_install_state.json`.